Removing a grain interface from the project fails the build

How to reproduce:

1. Create a new project for grain interfaces
2. Reference `Microsoft.Orleans.OrleansCodeGenerator.Build` `2.0.0-rc1` and `Microsoft.Orleans.Runtime.Abstractions` `2.0.0-rc1`
3. Add two grain interfaces
4. Build the project so that the code will be generated
5. Remove one interface and rebuild the project
6. Build fails with the following output
```
1>------ Rebuild All started: Project: DMCTS.GrainInterfaces, Configuration: Debug Any CPU ------
1>obj\Debug\netcoreapp2.0\DMCTS.GrainInterfaces.orleans.g.cs(19,305,19,317): error CS0234: The type or namespace name 'ITreeGrainV2' does not exist in the namespace 'DMCTS.GrainInterfaces' (are you missing an assembly reference?)
1>obj\Debug\netcoreapp2.0\DMCTS.GrainInterfaces.orleans.g.cs(87,192,87,204): error CS0234: The type or namespace name 'ITreeGrainV2' does not exist in the namespace 'DMCTS.GrainInterfaces' (are you missing an assembly reference?)
1>obj\Debug\netcoreapp2.0\DMCTS.GrainInterfaces.orleans.g.cs(20,127,20,139): error CS0234: The type or namespace name 'ITreeGrainV2' does not exist in the namespace 'DMCTS.GrainInterfaces' (are you missing an assembly reference?)
1>Done building project "DMCTS.GrainInterfaces.csproj" -- FAILED.
========== Rebuild All: 0 succeeded, 1 failed, 0 skipped ==========
```

Deleting `obj` directory helps.

--- a/src/Orleans.CodeGeneration.Build/build/Microsoft.Orleans.OrleansCodeGenerator.Build.targets
+++ b/src/Orleans.CodeGeneration.Build/build/Microsoft.Orleans.OrleansCodeGenerator.Build.targets
@@ -43,7 +43,7 @@
   <Target Name="GenerateOrleansCode"
           BeforeTargets="AssignTargetPaths"
           Condition="'$(CodeGeneratorEnabled)' == 'true'"
-          Inputs="@(Compile);@(ReferencePath)"
+          Inputs="@(Compile);@(ReferencePath);$(IntermediateOutputPath)$(MSBuildProjectFile).CoreCompileInputs.cache"
           Outputs="$(OutputFileName)">
     <PropertyGroup>
       <ExcludeCodeGen>$(DefineConstants);EXCLUDE_CODEGEN</ExcludeCodeGen>

--- a/src/Orleans.CodeGeneration.Build/build/Microsoft.Orleans.OrleansCodeGenerator.Build.targets
+++ b/src/Orleans.CodeGeneration.Build/build/Microsoft.Orleans.OrleansCodeGenerator.Build.targets
@@ -41,7 +41,8 @@
 
   <!-- This target is run just before Compile for an Orleans Grain Interface Project -->
   <Target Name="GenerateOrleansCode"
-          BeforeTargets="AssignTargetPaths"
+          AfterTargets="ResolveReferences"
+          BeforeTargets="CoreCompile"
           Condition="'$(CodeGeneratorEnabled)' == 'true'"
           Inputs="@(Compile);@(ReferencePath);$(IntermediateOutputPath)$(MSBuildProjectFile).CoreCompileInputs.cache"
           Outputs="$(OutputFileName)">

